server/wavelength/modules/economy/index.js

```diff
@@ -0,0 +1,180 @@
+// Used with nef
+
+'use strict';
+
+/** @type {typeof import('../../../../lib/fs').FS} */
+const FS = require('../../../../.lib-dist/fs').FS;
+const DB_PATH = 'config/database'; // path to your database folder
+
+module.exports = class Economy {
+	constructor(name, dbName) {
+		this.name = name;
+		this.dbName = dbName;
+		this.db = Db[dbName];
+
+		this.transactionLogs = FS('logs/transactions.txt');
+		this.tLogs = this.transactionLogs.createAppendStream();
+	}
+
+	/**
+	 * Returns the currency label
+	 * @param {number} value
+	 * @param {boolean} [includeValue=true]
+	 * @return {string}
+	 */
+
+	getCurrencyLabel(value, includeValue = true) {
+		return `${includeValue ? value : ''} ${this.name}${Chat.plural(value)}`;
+	}
+
+	/**
+	 * Returns the whether value is a positive integer or not
+	 * @param {number} value
+	 * @return {boolean}
+	 */
+
+	isPInteger(value) {
+		let numValue = Number(value);
+
+		return Number.isInteger(numValue) && numValue > 0;
+	}
+
+	/**
+	 * Returns a user's savings
+	 * @param {string} userid
+	 * @return {number}
+	 */
+
+	get(userid) {
+		return this.db.get(userid, 0);
+	}
+
+	/**
+	 * Awards users with currency
+	 * @param {string} target
+	 * @param {number} value
+	 * @return {string}
+	 */
+
+	award(target, value) {
+		if (!this.isPInteger(value)) return false;
+
+		let tarId = toId(target);
+
+		this.db.put(tarId, amount => amount + Number(value), 0);
+
+		let amount = this.get(tarId);
+
+		return this.getCurrencyLabel(amount);
+	}
+
+	/**
+	 * Removes a specified portion of user's savings
+	 * @param {string} target
+	 * @param {number} value
+	 * @return {string}
+	 */
+
+	remove(target, value) {
+		if (!this.isPInteger(value)) return false;
+
+		let tarId = toId(target);
+
+		this.db.put(tarId, amount => amount - Number(value), 0);
+
+		let amount = this.get(tarId);
+		if (amount <= 0) {
+			this.db.remove(tarId);
+			return `0 ${this.name}s`;
+		}
+
+		return this.getCurrencyLabel(amount);
+	}
+
+	/**
+	 * Returns the total currency, and the mean
+	 * @return {array}
```

Comment:
The return type is `[number, number]`.

Reply:
Would `number[]` be any different?

Reply:
Since there are a defined number of returned items (and only two of them at that), it's clearer to write `[number, number]`; however, `number[]` will do the same thing.